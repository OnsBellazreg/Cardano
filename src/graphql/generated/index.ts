import { gql } from '@apollo/client';
import * as Apollo from '@apollo/client';
export type Maybe<T> = T | null;
export type InputMaybe<T> = Maybe<T>;
export type Exact<T extends { [key: string]: unknown }> = { [K in keyof T]: T[K] };
export type MakeOptional<T, K extends keyof T> = Omit<T, K> & { [SubKey in K]?: Maybe<T[SubKey]> };
export type MakeMaybe<T, K extends keyof T> = Omit<T, K> & { [SubKey in K]: Maybe<T[SubKey]> };
export type MakeEmpty<T extends { [key: string]: unknown }, K extends keyof T> = { [_ in K]?: never };
export type Incremental<T> = T | { [P in keyof T]?: P extends ' $fragmentName' | '__typename' ? T[P] : never };
const defaultOptions = {} as const;
/** All built-in and custom scalars, mapped to their actual values */
export type Scalars = {
  ID: { input: string; output: string; }
  String: { input: string; output: string; }
  Boolean: { input: boolean; output: boolean; }
  Int: { input: number; output: number; }
  Float: { input: number; output: number; }
  Upload: { input: any; output: any; }
};

export enum CacheControlScope {
  Private = 'PRIVATE',
  Public = 'PUBLIC'
}

/** A `Lift` is a chairlift, gondola, tram, funicular, pulley, rope tow, or other means of ascending a mountain. */
export type Lift = {
  __typename?: 'Lift';
  /** The number of people that a `Lift` can hold */
  capacity: Scalars['Int']['output'];
  /** The number of feet in elevation that a `Lift` ascends */
  elevationGain: Scalars['Int']['output'];
  /** The unique identifier for a `Lift` (id: "panorama") */
  id: Scalars['ID']['output'];
  /** The name of a `Lift` */
  name: Scalars['String']['output'];
  /** A boolean describing whether a `Lift` is open for night skiing */
  night: Scalars['Boolean']['output'];
  /** The current status for a `Lift`: `OPEN`, `CLOSED`, `HOLD` */
  status?: Maybe<LiftStatus>;
  /** A list of trails that this `Lift` serves */
  trailAccess: Array<Trail>;
};

/** An enum describing the options for `LiftStatus`: `OPEN`, `CLOSED`, `HOLD` */
export enum LiftStatus {
  Closed = 'CLOSED',
  Hold = 'HOLD',
  Open = 'OPEN'
}

export type Mutation = {
  __typename?: 'Mutation';
  /** Sets a `Lift` status by sending `id` and `status` */
  setLiftStatus: Lift;
  /** Sets a `Trail` status by sending `id` and `status` */
  setTrailStatus: Trail;
};


export type MutationSetLiftStatusArgs = {
  id: Scalars['ID']['input'];
  status: LiftStatus;
};


export type MutationSetTrailStatusArgs = {
  id: Scalars['ID']['input'];
  status: TrailStatus;
};

export type Query = {
  __typename?: 'Query';
  /** Returns a `Lift` by `id` (id: "panorama") */
  Lift: Lift;
  /** Returns a `Trail` by `id` (id: "old-witch") */
  Trail: Trail;
  /** A list of all `Lift` objects */
  allLifts: Array<Lift>;
  /** A list of all `Trail` objects */
  allTrails: Array<Trail>;
  /** Returns an `Int` of `Lift` objects with optional `LiftStatus` filter */
  liftCount: Scalars['Int']['output'];
  /** Returns a list of `SearchResult` objects based on `term` or `status` */
  search: Array<SearchResult>;
  /** Returns an `Int` of `Trail` objects with optional `TrailStatus` filter */
  trailCount: Scalars['Int']['output'];
};


export type QueryLiftArgs = {
  id: Scalars['ID']['input'];
};


export type QueryTrailArgs = {
  id: Scalars['ID']['input'];
};


export type QueryAllLiftsArgs = {
  status?: InputMaybe<LiftStatus>;
};


export type QueryAllTrailsArgs = {
  status?: InputMaybe<TrailStatus>;
};


export type QueryLiftCountArgs = {
  status?: InputMaybe<LiftStatus>;
};


export type QuerySearchArgs = {
  status?: InputMaybe<LiftStatus>;
  term?: InputMaybe<Scalars['String']['input']>;
};


export type QueryTrailCountArgs = {
  status?: InputMaybe<TrailStatus>;
};

/** This union type returns one of two types: a `Lift` or a `Trail`. When we search for a letter, we'll return a list of either `Lift` or `Trail` objects. */
export type SearchResult = Lift | Trail;

export type Subscription = {
  __typename?: 'Subscription';
  /** Listens for changes in lift status */
  liftStatusChange?: Maybe<Lift>;
  /** Listens for changes in trail status */
  trailStatusChange?: Maybe<Trail>;
};

/** A `Trail` is a run at a ski resort */
export type Trail = {
  __typename?: 'Trail';
  /** A list of Lifts that provide access to this `Trail` */
  accessedByLifts: Array<Lift>;
  /** The difficulty rating for a `Trail` */
  difficulty: Scalars['String']['output'];
  /** A boolean describing whether or not a `Trail` is groomed */
  groomed: Scalars['Boolean']['output'];
  /** A unique identifier for a `Trail` (id: 'hemmed-slacks') */
  id: Scalars['ID']['output'];
  /** The name of a `Trail` */
  name: Scalars['String']['output'];
  /** A boolean describing whether or not a `Trail` is open for night skiing */
  night: Scalars['Boolean']['output'];
  /** The current status for a `Trail`: OPEN, CLOSED */
  status?: Maybe<TrailStatus>;
  /** A boolean describing whether or not a `Trail` has trees */
  trees: Scalars['Boolean']['output'];
};

/** An enum describing the options for `TrailStatus`: `OPEN`, `CLOSED` */
export enum TrailStatus {
  Closed = 'CLOSED',
  Open = 'OPEN'
}

export type AllTrailsQueryVariables = Exact<{ [key: string]: never; }>;


export type AllTrailsQuery = { __typename?: 'Query', allTrails: Array<{ __typename?: 'Trail', id: string, name: string, status?: TrailStatus | null, difficulty: string, groomed: boolean, trees: boolean, night: boolean, accessedByLifts: Array<{ __typename?: 'Lift', id: string, name: string, status?: LiftStatus | null, capacity: number, night: boolean, elevationGain: number }> }> };

export type TrailQueryVariables = Exact<{
  id: Scalars['ID']['input'];
}>;


export type TrailQuery = { __typename?: 'Query', Trail: { __typename?: 'Trail', id: string, name: string, status?: TrailStatus | null, difficulty: string, groomed: boolean, trees: boolean, night: boolean, accessedByLifts: Array<{ __typename?: 'Lift', id: string, name: string, status?: LiftStatus | null, capacity: number, night: boolean, elevationGain: number }> } };


export const AllTrailsDocument = gql`
    query AllTrails {
  allTrails {
    id
    name
    status
    difficulty
    groomed
    trees
    night
    accessedByLifts {
      id
      name
      status
      capacity
      night
      elevationGain
    }
  }
}
    `;

/**
 * __useAllTrailsQuery__
 *
 * To run a query within a React component, call `useAllTrailsQuery` and pass it any options that fit your needs.
 * When your component renders, `useAllTrailsQuery` returns an object from Apollo Client that contains loading, error, and data properties
 * you can use to render your UI.
 *
 * @param baseOptions options that will be passed into the query, supported options are listed on: https://www.apollographql.com/docs/react/api/react-hooks/#options;
 *
 * @example
 * const { data, loading, error } = useAllTrailsQuery({
 *   variables: {
 *   },
 * });
 */
export function useAllTrailsQuery(baseOptions?: Apollo.QueryHookOptions<AllTrailsQuery, AllTrailsQueryVariables>) {
        const options = {...defaultOptions, ...baseOptions}
        return Apollo.useQuery<AllTrailsQuery, AllTrailsQueryVariables>(AllTrailsDocument, options);
      }
export function useAllTrailsLazyQuery(baseOptions?: Apollo.LazyQueryHookOptions<AllTrailsQuery, AllTrailsQueryVariables>) {
          const options = {...defaultOptions, ...baseOptions}
          return Apollo.useLazyQuery<AllTrailsQuery, AllTrailsQueryVariables>(AllTrailsDocument, options);
        }
export function useAllTrailsSuspenseQuery(baseOptions?: Apollo.SuspenseQueryHookOptions<AllTrailsQuery, AllTrailsQueryVariables>) {
          const options = {...defaultOptions, ...baseOptions}
          return Apollo.useSuspenseQuery<AllTrailsQuery, AllTrailsQueryVariables>(AllTrailsDocument, options);
        }
export type AllTrailsQueryHookResult = ReturnType<typeof useAllTrailsQuery>;
export type AllTrailsLazyQueryHookResult = ReturnType<typeof useAllTrailsLazyQuery>;
export type AllTrailsSuspenseQueryHookResult = ReturnType<typeof useAllTrailsSuspenseQuery>;
export type AllTrailsQueryResult = Apollo.QueryResult<AllTrailsQuery, AllTrailsQueryVariables>;
export const TrailDocument = gql`
    query Trail($id: ID!) {
  Trail(id: $id) {
    id
    name
    status
    difficulty
    groomed
    trees
    night
    accessedByLifts {
      id
      name
      status
      capacity
      night
      elevationGain
    }
  }
}
    `;

/**
 * __useTrailQuery__
 *
 * To run a query within a React component, call `useTrailQuery` and pass it any options that fit your needs.
 * When your component renders, `useTrailQuery` returns an object from Apollo Client that contains loading, error, and data properties
 * you can use to render your UI.
 *
 * @param baseOptions options that will be passed into the query, supported options are listed on: https://www.apollographql.com/docs/react/api/react-hooks/#options;
 *
 * @example
 * const { data, loading, error } = useTrailQuery({
 *   variables: {
 *      id: // value for 'id'
 *   },
 * });
 */
export function useTrailQuery(baseOptions: Apollo.QueryHookOptions<TrailQuery, TrailQueryVariables> & ({ variables: TrailQueryVariables; skip?: boolean; } | { skip: boolean; }) ) {
        const options = {...defaultOptions, ...baseOptions}
        return Apollo.useQuery<TrailQuery, TrailQueryVariables>(TrailDocument, options);
      }
export function useTrailLazyQuery(baseOptions?: Apollo.LazyQueryHookOptions<TrailQuery, TrailQueryVariables>) {
          const options = {...defaultOptions, ...baseOptions}
          return Apollo.useLazyQuery<TrailQuery, TrailQueryVariables>(TrailDocument, options);
        }
export function useTrailSuspenseQuery(baseOptions?: Apollo.SuspenseQueryHookOptions<TrailQuery, TrailQueryVariables>) {
          const options = {...defaultOptions, ...baseOptions}
          return Apollo.useSuspenseQuery<TrailQuery, TrailQueryVariables>(TrailDocument, options);
        }
export type TrailQueryHookResult = ReturnType<typeof useTrailQuery>;
export type TrailLazyQueryHookResult = ReturnType<typeof useTrailLazyQuery>;
export type TrailSuspenseQueryHookResult = ReturnType<typeof useTrailSuspenseQuery>;
export type TrailQueryResult = Apollo.QueryResult<TrailQuery, TrailQueryVariables>;