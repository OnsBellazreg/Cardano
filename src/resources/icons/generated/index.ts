export { default as Logo } from './Logo';
export { default as Mountain } from './Mountain';
