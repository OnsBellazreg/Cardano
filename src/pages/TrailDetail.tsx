import {
  Badge,
  Box,
  Button,
  Card,
  CardBody,
  CardHeader,
  Divider,
  Flex,
  Heading,
  HStack,
  Icon,
  Image,
  SimpleGrid,
  StackDivider,
  Text,
  VStack,
} from '@chakra-ui/react';
import React from 'react';
import {
  FaAngleDoubleUp,
  FaArrowLeft,
  FaMoon,
  FaRegMoon,
  FaSkiing,
  FaSnowflake,
  FaTree,
  FaUserFriends,
} from 'react-icons/fa';
import { GrStatusUnknown } from 'react-icons/gr';
import { Link, redirect, useNavigate, useParams } from 'react-router-dom';

import TrailDetailSkeleton from '@/components/Trail/TrailDetail.skeleton';
import { Lift, useTrailQuery } from '@/graphql/generated';
import { getElevations } from '@/utils/trail.utils';

interface DetailItemProps {
  icon: React.ElementType;
  color: string;
  label: string;
  value: string;
  className?: string;
}

const getDifficultyColor = (difficulty: string) => {
  switch (difficulty.toLowerCase()) {
    case 'beginner':
      return 'green.500';
    case 'intermediate':
      return 'yellow.500';
    case 'advanced':
    case 'expert':
      return 'red.500';
    default:
      return 'gray.500';
  }
};

export const DetailItem = ({ icon, color, label, value }: DetailItemProps) => {
  return (
    <HStack>
      <Icon as={icon} color={color} />
      <Text>
        <strong>{label}:</strong> {value}
      </Text>
    </HStack>
  );
};

export const TrailDetail = () => {
  const { id } = useParams<{ id: string }>();
  const { data, loading, error } = useTrailQuery({
    variables: { id: id ?? 'unknown' },
  });

  if (loading) return <TrailDetailSkeleton />;
  if (error) return <p>Error: {error.message}</p>;
  const trail = data?.Trail;

  if (!trail) return <Text>Trail not found.</Text>;

  return (
    <Box maxWidth={'100%'} w={'container.lg'} m={'auto'} p={4}>
      {/* Back button */}
      <Button as={Link} to="/" aria-label="Go back" mb={4}>
        Go Back
      </Button>
      <SimpleGrid columns={{ base: 1, lg: 2 }} spacing={4}>
        <Flex
          position="relative"
          gap={2}
          alignItems="center"
          justifyContent="center"
        >
          <Image src={'/images/trail.png'} />
          <Flex
            position={'absolute'}
            top="50%"
            left="50%"
            transform={'translate(-50%, -50%)'}
            w={'100%'}
            h={'100%'}
            alignItems="center"
            justifyContent="center"
            background={'rgba(0, 0, 0, 0.3)'}
          >
            <Heading
              display="inline-block"
              color="white"
              as="h2"
              size="2xl"
              pr={2}
            >
              {trail.name}
            </Heading>
            <Badge
              fontSize="xl"
              colorScheme={
                trail.status === 'OPEN'
                  ? 'green'
                  : trail.status === 'CLOSED'
                    ? 'red'
                    : 'yellow'
              }
            >
              {trail.status}
            </Badge>
          </Flex>
        </Flex>
        <Box>
          <Heading as="h3" size="md" mb={4}>
            Details
          </Heading>
          <VStack align="start" spacing={2} divider={<StackDivider />}>
            <DetailItem
              data-testid="groomed-item"
              icon={FaSnowflake}
              color={trail.groomed ? 'teal.500' : 'gray.500'}
              label="Groomed"
              value={trail.groomed ? 'Yes' : 'No'}
            />
            <DetailItem
              icon={FaTree}
              color={trail.trees ? 'green.500' : 'gray.500'}
              label="Trees"
              value={trail.trees ? 'Yes' : 'No'}
            />
            <DetailItem
              icon={FaMoon}
              color={trail.night ? 'purple.500' : 'gray.500'}
              label="Night"
              value={trail.night ? 'Yes' : 'No'}
            />
            <DetailItem
              icon={FaSkiing}
              color={getDifficultyColor(trail.difficulty)}
              label="Difficulty"
              value={
                trail.difficulty.charAt(0).toUpperCase() +
                trail.difficulty.slice(1)
              }
            />
            <DetailItem
              icon={FaAngleDoubleUp}
              color="gray.500"
              label="Elevation Gain"
              value={`${getElevations(trail.accessedByLifts as Lift[])} ft`}
            />
          </VStack>
        </Box>
      </SimpleGrid>
      <Divider mt={4} mb={4} />
      <Box>
        <Heading as="h3" size="md" mb={4}>
          Accessed By Lifts
        </Heading>
        <SimpleGrid gap={4} columns={{ base: 1, sm: 2, md: 3, lg: 4 }}>
          {trail.accessedByLifts.map((lift) => (
            <Card key={lift.name}>
              <CardHeader borderBottom="1px" borderColor="gray.200">
                <Heading fontSize="medium" as={'h4'}>
                  {lift.name}
                </Heading>
              </CardHeader>
              <CardBody>
                <VStack align="start" spacing={2} divider={<StackDivider />}>
                  <DetailItem
                    color="gray.500"
                    icon={GrStatusUnknown}
                    label="Status"
                    value={lift.status as string}
                  />
                  <DetailItem
                    color="gray.500"
                    icon={FaRegMoon}
                    label="Night"
                    value={lift.night ? 'Yes' : 'No'}
                  />
                  <DetailItem
                    color="gray.500"
                    icon={FaUserFriends}
                    label="Capacity"
                    value={`${lift.capacity}`}
                  />
                  <DetailItem
                    color="gray.500"
                    icon={FaAngleDoubleUp}
                    label="Elevation"
                    value={`${lift.elevationGain} ft`}
                  />
                </VStack>
              </CardBody>
            </Card>
          ))}
        </SimpleGrid>
      </Box>
    </Box>
  );
};

export default TrailDetail;
