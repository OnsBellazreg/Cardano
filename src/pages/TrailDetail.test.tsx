import { render, screen } from '@testing-library/react';
import React from 'react';
import { BrowserRouter as Router, useParams } from 'react-router-dom';

import { Lift, LiftStatus, Trail, useTrailQuery } from '@/graphql/generated';

import { TrailDetail } from './TrailDetail';

// Mock the necessary hooks
jest.mock('@/graphql/generated', () => ({
  useTrailQuery: jest.fn(),
  useParams: jest.fn(),
  LiftStatus: {
    Open: 'OPEN',
    Closed: 'CLOSED',
    Hold: 'HOLD',
  },
  TrailStatus: {
    Open: 'OPEN',
    Closed: 'CLOSED',
  },
}));

const mockLift: Lift = {
  id: 'lift-1',
  name: 'Lift A',
  status: LiftStatus.Open,
  night: true,
  capacity: 4,
  elevationGain: 300,
  trailAccess: [],
};

const mockTrail: Trail = {
  id: '1',
  name: 'Blue Bird',
  groomed: false,
  status: undefined,
  accessedByLifts: [mockLift],
  difficulty: 'Easy',
  night: true,
  trees: false,
};

const MockTrailDetailWithParams = () => {
  const { id } = useParams<{ id: string }>();
  return <TrailDetail />;
};

describe('TrailDetail', () => {
  beforeEach(() => {
    (useTrailQuery as jest.Mock).mockReturnValue({
      data: { Trail: mockTrail },
      loading: false,
      error: null,
    });
  });

  // it('renders the loading skeleton while data is loading', () => {
  //   (useTrailQuery as jest.Mock).mockReturnValue({
  //     data: { allTrails: [] },
  //     loading: true,
  //     error: null,
  //   });
  //   const { container } = render(<TrailDetail />);
  //   expect(container).toMatchSnapshot();
  // });

  it('displays trail details when data is loaded', async () => {
    const { container } = render(<MockTrailDetailWithParams />);
    expect(container).toMatchSnapshot();
  });

  it('handles error when data fetching fails', async () => {
    (useTrailQuery as jest.Mock).mockReturnValue({
      data: { Trail: null },
      loading: false,
      error: { message: 'Failed to fetch trail details' },
    });

    const { container } = render(<MockTrailDetailWithParams />);

    expect(
      await screen.findByText('Error: Failed to fetch trail details'),
    ).toBeInTheDocument();

    expect(container).toMatchSnapshot();
  });

  it('shows Trail not found message when no trail matches the ID', async () => {
    (useTrailQuery as jest.Mock).mockReturnValue({
      data: { Trail: null },
      loading: false,
      error: null,
    });

    const { container } = render(<MockTrailDetailWithParams />);
    expect(container).toMatchSnapshot();

    expect(await screen.findByText('Trail not found.')).toBeInTheDocument()
});