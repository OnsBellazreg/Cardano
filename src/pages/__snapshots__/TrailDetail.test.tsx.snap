// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TrailDetail displays trail details when data is loaded 1`] = `
<div>
  <div
    class="css-iab5yi"
  >
    <a
      aria-label="Go back"
      class="chakra-button css-eaw8jh"
      href="/"
    >
      Go Back
    </a>
    <div
      class="css-13t9dym"
    >
      <div
        class="css-z3npd7"
      >
        <img
          class="chakra-image css-0"
          src="/images/trail.png"
        />
        <div
          class="css-tc8c76"
        >
          <h2
            class="chakra-heading css-1uvsvno"
          >
            Blue Bird
          </h2>
          <span
            class="chakra-badge css-1m09api"
          />
        </div>
      </div>
      <div
        class="css-0"
      >
        <h3
          class="chakra-heading css-1itv5e3"
        >
          Details
        </h3>
        <div
          class="chakra-stack css-13kdors"
        >
          <div
            class="chakra-stack css-1igwmid"
          >
            <svg
              class="chakra-icon css-2u1jce"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <g
                stroke="currentColor"
                stroke-width="1.5"
              >
                <path
                  d="M9,9a3,3,0,1,1,4,2.829,1.5,1.5,0,0,0-1,1.415V14.25"
                  fill="none"
                  stroke-linecap="round"
                />
                <path
                  d="M12,17.25a.375.375,0,1,0,.375.375A.375.375,0,0,0,12,17.25h0"
                  fill="currentColor"
                  stroke-linecap="round"
                />
                <circle
                  cx="12"
                  cy="12"
                  fill="none"
                  r="11.25"
                  stroke-miterlimit="10"
                />
              </g>
            </svg>
            <p
              class="chakra-text css-0"
            >
              <strong>
                Groomed
                :
              </strong>
               
              No
            </p>
          </div>
          <div
            class="chakra-stack__divider css-xgvkl8"
          />
          <div
            class="chakra-stack css-1igwmid"
          >
            <svg
              class="chakra-icon css-2u1jce"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <g
                stroke="currentColor"
                stroke-width="1.5"
              >
                <path
                  d="M9,9a3,3,0,1,1,4,2.829,1.5,1.5,0,0,0-1,1.415V14.25"
                  fill="none"
                  stroke-linecap="round"
                />
                <path
                  d="M12,17.25a.375.375,0,1,0,.375.375A.375.375,0,0,0,12,17.25h0"
                  fill="currentColor"
                  stroke-linecap="round"
                />
                <circle
                  cx="12"
                  cy="12"
                  fill="none"
                  r="11.25"
                  stroke-miterlimit="10"
                />
              </g>
            </svg>
            <p
              class="chakra-text css-0"
            >
              <strong>
                Trees
                :
              </strong>
               
              No
            </p>
          </div>
          <div
            class="chakra-stack__divider css-xgvkl8"
          />
          <div
            class="chakra-stack css-1igwmid"
          >
            <svg
              class="chakra-icon css-1667dh9"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <g
                stroke="currentColor"
                stroke-width="1.5"
              >
                <path
                  d="M9,9a3,3,0,1,1,4,2.829,1.5,1.5,0,0,0-1,1.415V14.25"
                  fill="none"
                  stroke-linecap="round"
                />
                <path
                  d="M12,17.25a.375.375,0,1,0,.375.375A.375.375,0,0,0,12,17.25h0"
                  fill="currentColor"
                  stroke-linecap="round"
                />
                <circle
                  cx="12"
                  cy="12"
                  fill="none"
                  r="11.25"
                  stroke-miterlimit="10"
                />
              </g>
            </svg>
            <p
              class="chakra-text css-0"
            >
              <strong>
                Night
                :
              </strong>
               
              Yes
            </p>
          </div>
          <div
            class="chakra-stack__divider css-xgvkl8"
          />
          <div
            class="chakra-stack css-1igwmid"
          >
            <svg
              class="chakra-icon css-2u1jce"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <g
                stroke="currentColor"
                stroke-width="1.5"
              >
                <path
                  d="M9,9a3,3,0,1,1,4,2.829,1.5,1.5,0,0,0-1,1.415V14.25"
                  fill="none"
                  stroke-linecap="round"
                />
                <path
                  d="M12,17.25a.375.375,0,1,0,.375.375A.375.375,0,0,0,12,17.25h0"
                  fill="currentColor"
                  stroke-linecap="round"
                />
                <circle
                  cx="12"
                  cy="12"
                  fill="none"
                  r="11.25"
                  stroke-miterlimit="10"
                />
              </g>
            </svg>
            <p
              class="chakra-text css-0"
            >
              <strong>
                Difficulty
                :
              </strong>
               
              Easy
            </p>
          </div>
          <div
            class="chakra-stack__divider css-xgvkl8"
          />
          <div
            class="chakra-stack css-1igwmid"
          >
            <svg
              class="chakra-icon css-2u1jce"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <g
                stroke="currentColor"
                stroke-width="1.5"
              >
                <path
                  d="M9,9a3,3,0,1,1,4,2.829,1.5,1.5,0,0,0-1,1.415V14.25"
                  fill="none"
                  stroke-linecap="round"
                />
                <path
                  d="M12,17.25a.375.375,0,1,0,.375.375A.375.375,0,0,0,12,17.25h0"
                  fill="currentColor"
                  stroke-linecap="round"
                />
                <circle
                  cx="12"
                  cy="12"
                  fill="none"
                  r="11.25"
                  stroke-miterlimit="10"
                />
              </g>
            </svg>
            <p
              class="chakra-text css-0"
            >
              <strong>
                Elevation Gain
                :
              </strong>
               
              300 ft
            </p>
          </div>
        </div>
      </div>
    </div>
    <hr
      aria-orientation="horizontal"
      class="chakra-divider css-pwzhxe"
    />
    <div
      class="css-0"
    >
      <h3
        class="chakra-heading css-1itv5e3"
      >
        Accessed By Lifts
      </h3>
      <div
        class="css-1d3vlji"
      >
        <div
          class="chakra-card css-2dlz7"
        >
          <div
            class="chakra-card__header css-1g27vng"
          >
            <h4
              class="chakra-heading css-ex1lad"
            >
              Lift A
            </h4>
          </div>
          <div
            class="chakra-card__body css-0"
          >
            <div
              class="chakra-stack css-13kdors"
            >
              <div
                class="chakra-stack css-1igwmid"
              >
                <svg
                  class="chakra-icon css-2u1jce"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <g
                    stroke="currentColor"
                    stroke-width="1.5"
                  >
                    <path
                      d="M9,9a3,3,0,1,1,4,2.829,1.5,1.5,0,0,0-1,1.415V14.25"
                      fill="none"
                      stroke-linecap="round"
                    />
                    <path
                      d="M12,17.25a.375.375,0,1,0,.375.375A.375.375,0,0,0,12,17.25h0"
                      fill="currentColor"
                      stroke-linecap="round"
                    />
                    <circle
                      cx="12"
                      cy="12"
                      fill="none"
                      r="11.25"
                      stroke-miterlimit="10"
                    />
                  </g>
                </svg>
                <p
                  class="chakra-text css-0"
                >
                  <strong>
                    Status
                    :
                  </strong>
                   
                  OPEN
                </p>
              </div>
              <div
                class="chakra-stack__divider css-xgvkl8"
              />
              <div
                class="chakra-stack css-1igwmid"
              >
                <svg
                  class="chakra-icon css-2u1jce"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <g
                    stroke="currentColor"
                    stroke-width="1.5"
                  >
                    <path
                      d="M9,9a3,3,0,1,1,4,2.829,1.5,1.5,0,0,0-1,1.415V14.25"
                      fill="none"
                      stroke-linecap="round"
                    />
                    <path
                      d="M12,17.25a.375.375,0,1,0,.375.375A.375.375,0,0,0,12,17.25h0"
                      fill="currentColor"
                      stroke-linecap="round"
                    />
                    <circle
                      cx="12"
                      cy="12"
                      fill="none"
                      r="11.25"
                      stroke-miterlimit="10"
                    />
                  </g>
                </svg>
                <p
                  class="chakra-text css-0"
                >
                  <strong>
                    Night
                    :
                  </strong>
                   
                  Yes
                </p>
              </div>
              <div
                class="chakra-stack__divider css-xgvkl8"
              />
              <div
                class="chakra-stack css-1igwmid"
              >
                <svg
                  class="chakra-icon css-2u1jce"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <g
                    stroke="currentColor"
                    stroke-width="1.5"
                  >
                    <path
                      d="M9,9a3,3,0,1,1,4,2.829,1.5,1.5,0,0,0-1,1.415V14.25"
                      fill="none"
                      stroke-linecap="round"
                    />
                    <path
                      d="M12,17.25a.375.375,0,1,0,.375.375A.375.375,0,0,0,12,17.25h0"
                      fill="currentColor"
                      stroke-linecap="round"
                    />
                    <circle
                      cx="12"
                      cy="12"
                      fill="none"
                      r="11.25"
                      stroke-miterlimit="10"
                    />
                  </g>
                </svg>
                <p
                  class="chakra-text css-0"
                >
                  <strong>
                    Capacity
                    :
                  </strong>
                   
                  4
                </p>
              </div>
              <div
                class="chakra-stack__divider css-xgvkl8"
              />
              <div
                class="chakra-stack css-1igwmid"
              >
                <svg
                  class="chakra-icon css-2u1jce"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <g
                    stroke="currentColor"
                    stroke-width="1.5"
                  >
                    <path
                      d="M9,9a3,3,0,1,1,4,2.829,1.5,1.5,0,0,0-1,1.415V14.25"
                      fill="none"
                      stroke-linecap="round"
                    />
                    <path
                      d="M12,17.25a.375.375,0,1,0,.375.375A.375.375,0,0,0,12,17.25h0"
                      fill="currentColor"
                      stroke-linecap="round"
                    />
                    <circle
                      cx="12"
                      cy="12"
                      fill="none"
                      r="11.25"
                      stroke-miterlimit="10"
                    />
                  </g>
                </svg>
                <p
                  class="chakra-text css-0"
                >
                  <strong>
                    Elevation
                    :
                  </strong>
                   
                  300 ft
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`TrailDetail handles error when data fetching fails 1`] = `
<div>
  <p>
    Error: 
    Failed to fetch trail details
  </p>
</div>
`;

exports[`TrailDetail shows Trail not found message when no trail matches the ID 1`] = `
<div>
  <p
    class="chakra-text css-0"
  >
    Trail not found.
  </p>
</div>
`;
